projects/m1/016-next-day/solutions/stefanoconsonni/solution.js

```diff
@@ -0,0 +1,44 @@
+let enteredYear = prompt("Please enter a year (you must enter four digits):");
+let enteredMonth = prompt("Please enter a month (you must enter one or two digits):");
+let enteredDay = prompt("Please enter a day (you must enter one or two digits):");
+
+let enteredDate = `${enteredYear}-${enteredMonth}-${enteredDay}`;
+
+let isLeapYear;
+if (enteredYear % 400 == 0) {
+	isLeapYear = true;
+} else if (enteredYear % 400 != 0 && enteredYear % 100 == 0) {
+	isLeapYear = false;
+} else if (enteredYear % 400 != 0 && enteredYear % 100 != 0 && enteredYear % 4 == 0) {
+	isLeapYear = true;
+} else {
+	isLeapYear = false;
+}
+
+let monthOf30Days;
+if (enteredMonth == 4 || enteredMonth == 6 || enteredMonth == 9 || enteredMonth == 11) {
+	monthOf30Days = true;
+} else {
+	monthOf30Days = false;
+}
+
+let monthOf31Days;
+if (enteredMonth == 1 || enteredMonth == 3 || enteredMonth == 5 || enteredMonth == 7 || enteredMonth == 8 || enteredMonth == 10 || enteredMonth == 12) {
+	monthOf31Days = true;
+} else {
+	monthOf31Days = false;
+}
+
+if (enteredMonth == 12 && enteredDay == 31) {
```

Comment:
Ci sono oggettivamente troppi controlli in questa porzione di codice, prova a rivederla cercando di comprimerla ulteriormente.

Reply:
Ciao Simone, ti ringrazio per il feedback! Ho modificato il codice e pushato il file js modificato, puoi dirmi se va meglio ora? Grazie